cloud-gov/caulking
# Update to version 6 of gitleaks

## Changes proposed in this pull request:

- Add a VERSION to the output for `make audit`
- Update README on how to run specific tests in bats files
- Added tests for detecting forbidden filenames
- Moved all the regexes exceptions for `regex = '''CHANGEME|cha` to a final `regexes` allowlist (support now in v6)
- Changed all the 'whitelists' to 'allowlists'
- Adopted the v6 configuration changes described at: https://github.com/zricethezav/gitleaks/releases/tag/v6.0.0
- Comments why we skip account number test

To test this:

- Run `brew upgrade gitleaks` and ensure with `gitleaks --version` you have v6.1.2
  - Note that `gitleaks --version` with older versions may fail
- run the following:

```sh
make patterns
bats development.bats # all should pass
bats caulked.bats  # all should pass
make audit # should pass, unless you're in violation
```


## security considerations

OK. Doesn't weaken any rules, doesn't reveal anything sensitive
